src/NuGet.Core/NuGet.Packaging/PublicAPI/net472/PublicAPI.Unshipped.txt

```diff
@@ -1 +1,6 @@
 #nullable enable
+NuGet.Packaging.Rules.MissingReadmeRule
+~NuGet.Packaging.Rules.MissingReadmeRule.MessageFormat.get -> string
+~NuGet.Packaging.Rules.MissingReadmeRule.MissingReadmeRule(string messageFormat) -> void
+~NuGet.Packaging.Rules.MissingReadmeRule.Validate(NuGet.Packaging.PackageArchiveReader builder) -> System.Collections.Generic.IEnumerable<NuGet.Common.PackagingLogMessage>
+~static NuGet.Packaging.Rules.AnalysisResources.MissingReadmeInformation.get -> string
```

Comment:
Are all rules public? From an API design point of view, I guess it might be useful if someone using the Client SDK wants to run a specific rule on a package, without running all other rules. But it also feels a bit overkill.

However, if we do want to keep it public, I don't think the constructor makes sense. I guess if someone calling the API from their own app wants a translation of the message that we don't provide ourselves it might be useful, but the XML doc for the constructor doesn't say how many parameters (`{x}`) the message takes, or what each of those substitutions mean. If the message is passed into the constructor, I guess maybe it makes sense to provide a property getting to get the value back again. But if I was writing an app using the API, I think only the `Validate(..)` method is useful. So, I think we can simplify the public API if we remove the getter.

Reply:
Another thing is that this newly added rule, contrary to all the other rules, does not really raise a warning. 

Do we need different sets for both rules?

Reply:
Thanks for the suggestion! @zivkan 
There are many internal rules, and several public rules.
So I changed it into internal instead.

Thanks for reviewing @nkolev92 !
I created a subfolder BestPractice under Rules folder. All rules need to implement the IPackageRule interface, this might be an easy way to differentiate the two sets of rules.